server/src/Application/Validation/Validators/Course/CourseValidationExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using FluentValidation;
+using StudyZen.Application.Services;
+using StudyZen.Domain.Constraints;
+
+namespace StudyZen.Application.Validation;
+
+public static class CourseValidationExtensions
+{
+    public static IRuleBuilderOptions<T, string?> CourseName<T>(this IRuleBuilder<T, string?> ruleBuilder)
+    {
+        return ruleBuilder
+            .MustNotBeNullOrWhitespace()
+            .LengthMustNotExceed(CourseConstraints.NameMaxLength);
+    }
+
+    public static IRuleBuilderOptions<T, string?> CourseDescription<T>(this IRuleBuilder<T, string?> ruleBuilder)
+    {
+        return ruleBuilder
+            .MustNotBeNull()
+            .LengthMustNotExceed(CourseConstraints.DescriptionMaxLength);
+    }
+
+    public static IRuleBuilderOptions<T, int> CourseId<T>(this IRuleBuilder<T, int> ruleBuilder, ICourseService courseService)
+    {
+        return ruleBuilder
+            .Must(id => courseService.GetCourseById(id) is not null)
+            .WithErrorCode(ValidationErrorCodes.NotFound);
```

Comment:
The current error message for this type of validation is generic, but as I understood from the description, this will be fixed in the future.